neopixel.py

```diff
@@ -170,13 +176,14 @@ def brightness(self):
 
     @brightness.setter
     def brightness(self, brightness):
+        # pylint: disable=attribute-defined-outside-init
         self._brightness = min(max(brightness, 0.0), 1.0)
 
     def fill(self, color):
         """Colors all pixels the given ***color***."""
         auto_write = self.auto_write
         self.auto_write = False
-        for i in range(len(self)):
+        for i, _ in enumerate(self):
```

Comment:
Be careful, enumerate needs to be explicitly turned on for certain MicroPython builds (with a config header bit change).  The safest and most universal code here is iterating with range.